### PR TITLE
AMPI+CUDA: fix build

### LINKED ISSUE #2427: <cuda-runtime.h> not found while building netlrts-linux-x86_64-cuda builds
Autobuild log: http://charm.cs.illinois.edu/autobuild/old.2019_07_25__01_04/netlrts-linux-x86_64-cuda.txt
```
In file included from ampi.h:11:0,
                 from compat_ampius.C:2:
/scratch/autobuild/netlrts-linux-x86_64-cuda/charm/netlrts-linux-x86_64-cuda/bin/../include/hapi.h:3:26: fatal error: cuda_runtime.h: No such file or directory
 #include <cuda_runtime.h>
                          ^
compilation terminated.
/bin/cp -f sim.h ../../../../include/pose/sim.h
```

### PR DESCRIPTION
**Pull request by matthiasdiener**
Fixes #2427. Followup from #2062/#2121. See also #2459. Only lightly tested.

### CODE REVIEW

File: src/arch/common/special.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ export CHARMINC=../include
 if test -n "$BUILD_CUDA"
 then
   echo "---------------------- special.sh for cuda executing ----------------"
-  if [ "$CMK_GDIR" == "gni" ] ; then
+  if [ "$CMK_GDIR" = "gni" ] ; then
```
Comment (minitu) [inline src/arch/common/special.sh]: Shouldn't this stay as double equal signs?

Comment (matthiasdiener) [reply src/arch/common/special.sh]: Double equal signs are only valid in bash’s “[[“ operator, not in “[“.

**Comment by evan-charmworks** [comment]
Getting the following spam:

```
/Users/evan/charm/netlrts-darwin-x86_64-smp/bin/charmc: line 1423: [: : integer expression expected
/Users/evan/charm/netlrts-darwin-x86_64-smp/bin/ampicc: line 83: [: -eq: unary operator expected
```
The latter line needs `""` around `$BUILD_CUDA`, but both fail because BUILD_CUDA is empty when off.

**Comment by matthiasdiener** [comment]
> Getting the following spam:
> 
> ```
> /Users/evan/charm/netlrts-darwin-x86_64-smp/bin/charmc: line 1423: [: : integer expression expected
> /Users/evan/charm/netlrts-darwin-x86_64-smp/bin/ampicc: line 83: [: -eq: unary operator expected
> ```
> 
> The latter line needs `""` around `$BUILD_CUDA`, but both fail because BUILD_CUDA is empty when off.

#2462 should fix this.

**Pull request merged**